homeassistant/components/touchline_sl/climate.py

```diff
@@ -0,0 +1,126 @@
+"""Roth Touchline SL climate integration implementation for Home Assistant."""
+
+from typing import Any
+
+from pytouchlinesl import Zone
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import TouchlineSLConfigEntry
+from .const import DOMAIN
+from .coordinator import TouchlineSLModuleCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: TouchlineSLConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Touchline devices."""
+    coordinators = entry.runtime_data
+    async_add_entities(
+        TouchlineSLZone(coordinator=coordinator, zone_id=zone_id)
+        for coordinator in coordinators
+        for zone_id in coordinator.data.zones
+    )
+
+
+CONST_TEMP_PRESET_NAME = "Constant Temperature"
+
+
+class TouchlineSLZone(CoordinatorEntity[TouchlineSLModuleCoordinator], ClimateEntity):
+    """Roth Touchline SL Zone."""
+
+    _attr_has_entity_name = True
+    _attr_hvac_mode = HVACMode.HEAT
+    _attr_hvac_modes = [HVACMode.HEAT]
+    _attr_name = None
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+
+    def __init__(self, coordinator: TouchlineSLModuleCoordinator, zone_id: int) -> None:
+        """Construct a Touchline SL climate zone."""
+        super().__init__(coordinator)
+        self.zone_id: int = zone_id
+
+        self._attr_unique_id = (
+            f"module-{self.coordinator.data.module.id}-zone-{self.zone_id}"
+        )
+
+        zone_name = coordinator.data.zones[zone_id].name
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, str(zone_id))},
+            name=zone_name,
+            manufacturer="Roth",
+            via_device=(DOMAIN, coordinator.data.module.id),
+            model="zone",
+            suggested_area=zone_name,
+        )
+
+        # Call this in __init__ so data is populated right away, since it's
+        # already available in the coordinator data.
+        self.set_attr()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self.set_attr()
+        super()._handle_coordinator_update()
+
+    @property
+    def device(self) -> Zone:
```

Comment:
```suggestion
    def zone(self) -> Zone:
```
Might as well name it zone